addons/account/models/account_move.py

```diff
@@ -524,8 +524,6 @@ def _onchange_partner_id(self):
                 line.account_id = new_term_account
 
         self._compute_bank_partner_id()
-        bank_ids = self.bank_partner_id.bank_ids.filtered(lambda bank: bank.company_id is False or bank.company_id == self.company_id)
-        self.partner_bank_id = bank_ids and bank_ids[0]
```

Comment:
Just a note:
Normally you would need to call `self._compute_partner_bank_id()` here.
This time it is not necessary because, for reasons, it will be called anyway when acting on the fiscal position, a few lines later